kafka-plugins-0.10/src/test/java/io/cdap/plugin/KafkaStreamingSourceTest.java

```diff
@@ -218,6 +220,88 @@ public void testKafkaStreamingSource() throws Exception {
     sparkManager.stop();
   }
 
+  @Test
+  public void testKafkaStreamingSourceInitialOffset() throws Exception {
+    Schema schema = Schema.recordOf(
+      "user",
+      Schema.Field.of("id", Schema.of(Schema.Type.LONG)),
+      Schema.Field.of("first", Schema.of(Schema.Type.STRING)),
+      Schema.Field.of("last", Schema.of(Schema.Type.STRING)));
+
+    Map<String, String> properties = new HashMap<>();
+    properties.put("referenceName", "kafkaPurchasesOffsetTest");
+    properties.put("brokers", "localhost:" + kafkaPort);
+    properties.put("topic", "usersWithOffset");
+    properties.put("initialOffset", OFFSET_START_FROM_LAST_OFFSET);
```

Comment:
can you also add tests for other values of initialOffset?

Reply:
updated tests